### DISCUSSION TITLE
Artifact id of pomless aggregators should be configurable.

### DISCUSSION THREAD

**Discussion by ghost** [eclipse-tycho/tycho#597] [Q&A] [answered]
A normal Eclipse project usually has the form:

```
root
+- plugins
|   +- plugin1
|   +- plugin2
|   +- ...
|
+- features
    +- feature1
    +- ...
```

When performing a pomless build of those bundles, Tycho creates polygot files in the plugins and features directories, with the group id inherited from the parent pom and the artifact id derived from the folder name.

Due to the size of our application, we have deviated from this structure by splitting features and plugins into separate, loosely related components. So our workspace looks something like this:

```
project
 +- component1
 |   +- plugins
 |   +- features
 |
 +- component2
     +- plugins
     +- features
```

In a pomless build, this will result in the aggregator poms of both components to share the same artifact id, making it impossible to build the project, unless the group id is overwritten for each component.

I think it would be great to have more control over how the artifact id is generated. My idea is something like a system property (e.g. "tycho.pomless.aggregator.depth"), which derives a name from the current directory, as well as their parent directories. The default depth should be 0, in order to replicate the current behavior.

Example id for different depths:
depth = 0 -> features
depth = 1 -> component1-features
depth = 2 -> project-component1-features

**Comment by laeubi** [comment, marked as answer]
You can create a build.properties file and there you can define a custom group id e.g.:

```
pom.model.groupId=my.group.id
pom.model.artifactId=my.custom.artifact.id
```

by the way: I think it would be good to describe this at the wiki so maybe you want to try this out and write a description here
https://github.com/eclipse/tycho/wiki/Tycho-Pomless

**Comment by ghost** [comment]
That's exactly it! Thank you a lot for the quick response :)
I'll create a new entry in the wiki as well.
